### PR TITLE
IGNITE-16201 JDBC: Batch execution

### PR DESCRIPTION
**Pull request by vladErmakov07**

### CODE REVIEW

File: modules/client-common/src/main/java/org/apache/ignite/client/proto/query/event/BatchExecuteResult.java [hunk 1]
```diff
@@ -30,12 +30,30 @@
...
 
+    /**
+     * Constructor.
+     *
+     * @param status Status code.
+     * @param errorCode Product error code.
+     * @param errorMessage Error message.
+     * @param updateCnts Array with success and error markers.
+     */
+    public BatchExecuteResult(int status, int errorCode, String errorMessage, int[] updateCnts) {
```
Comment (AMashenkov) [inline modules/client-common/src/main/java/org/apache/ignite/client/proto/query/event/BatchExecuteResult.java:50]: The constructor looks ambiguous. 
Is it possible to get a error and result at the same time?

File: modules/client-handler/src/main/java/org/apache/ignite/client/handler/JdbcQueryEventHandlerImpl.java [hunk 2]
```diff
@@ -164,8 +175,58 @@ public JdbcQueryEventHandlerImpl(QueryProcessor processor, JdbcMetadataCatalog m
...
+            }
+        } catch (Exception e) {
+            StringWriter sw = getWriterWithStackTrace(e);
+
+            return CompletableFuture.completedFuture(new BatchExecuteResult(Response.STATUS_FAILED,
+                    "Exception while executing query " + req.queries().get(0).sql() + ". Error message: " + sw));
+        }
+
+        if (!firstError.isEmpty()) {
+            return CompletableFuture.completedFuture(new BatchExecuteResult(
```
Comment (AMashenkov) [inline modules/client-handler/src/main/java/org/apache/ignite/client/handler/JdbcQueryEventHandlerImpl.java:199]: How to understand what query from the batch has failed if the 'res' contains results for all of the queries?

Comment (vladErmakov07) [reply modules/client-handler/src/main/java/org/apache/ignite/client/handler/JdbcQueryEventHandlerImpl.java:198]: The array element for any command that failed is Statement.EXECUTE_FAILED (-3).
On the other hand, there can only be one error, and JDBC driver contract does not regulate it for the case when several queries are failed.

My implementation is the same as the one in Ignite 2: only the error of the first broken query is sent to the client. The rest of the errors are logged on the server node. 
Maybe we should discuss this behavior.

Comment (vladErmakov07) [reply modules/client-handler/src/main/java/org/apache/ignite/client/handler/JdbcQueryEventHandlerImpl.java]: I refactored the exception handling. We won't continue with operations if the execution of a query ended with an error. Now, the result will always contain only positive results of update operations.

updateCounts – an update count or Statement.SUCCESS_NO_INFO for each SQL command in the batch prior to the failure for JDBC drivers that stop processing after a command failure

File: modules/client-common/src/main/java/org/apache/ignite/client/proto/query/event/BatchExecuteRequest.java [hunk 3]
```diff
@@ -93,8 +93,8 @@ public void writeBinary(ClientMessagePacker packer) {
 
         packer.packArrayHeader(queries.size());
 
-        for (Query q : queries) {
-            q.writeBinary(packer);
+        for (String q : queries) {
+            ClientMessageUtils.writeStringNullable(packer, q);
```
Comment (AMashenkov) [inline modules/client-common/src/main/java/org/apache/ignite/client/proto/query/event/BatchExecuteRequest.java:97]: Null string make no sense, isn't it?

Comment (vladErmakov07) [reply modules/client-common/src/main/java/org/apache/ignite/client/proto/query/event/BatchExecuteRequest.java]: Removed and added not null check. Thnx.

**Pull request merged**